src/core/io/IB_fbt.h

```
@@ -0,0 +1,100 @@
+    static const CStringDictionary::TStringId scmEventInputTypeIds[];
+    static const TEventID scmEventINITOID = 0;
+    static const TEventID scmEventCNFID = 1;
+    static const TEventID scmEventINDID = 2;
+    static const TDataIOID scmEOWith[];
+    static const TForteInt16 scmEOWithIndexes[];
+    static const CStringDictionary::TStringId scmEventOutputNames[];
+    static const CStringDictionary::TStringId scmEventOutputTypeIds[];
+
+    static const SFBInterfaceSpec scmFBInterfaceSpec;
+
+    void executeEvent(TEventID paEIID, CEventChainExecutionThread *const paECET) override;
+
+    void writeOutputData(TEventID paEIID) override;
+    void setInitialValues() override;
+
+    CIEC_BOOL read() override {
+      return forte::core::io::CProcessInterfaceFB::read(var_IN);
+    }
+
+  public:
+    FORTE_IB(CStringDictionary::TStringId paInstanceNameId, forte::core::CFBContainer &paContainer);
+
+    CIEC_BYTE var_IN;
+
+    CIEC_BYTE var_conn_IN;
+
+    CEventConnection conn_IND;
+
+    CDataConnection conn_IN;
+
+    CIEC_ANY *getDO(size_t) override;
+    CEventConnection *getEOConUnchecked(TPortId) override;
+    CDataConnection *getDOConUnchecked(TPortId) override;
+
+    void evt_INIT(const CIEC_BOOL &paQI, const CIEC_STRING &paPARAMS, CIEC_BOOL &paQO, CIEC_STRING &paSTATUS, CIEC_BYTE &paIN) {
+      var_QI = paQI;
+      var_PARAMS = paPARAMS;
+      receiveInputEvent(scmEventINITID, nullptr);
+      paQO = var_QO;
+      paSTATUS = var_STATUS;
+      paIN = var_IN;
+    }
+
+    void evt_REQ(const CIEC_BOOL &paQI, const CIEC_STRING &paPARAMS, CIEC_BOOL &paQO, CIEC_STRING &paSTATUS, CIEC_BYTE &paIN) {
+      var_QI = paQI;
+      var_PARAMS = paPARAMS;
+      receiveInputEvent(scmEventREQID, nullptr);
+      paQO = var_QO;
+      paSTATUS = var_STATUS;
+      paIN = var_IN;
+    }
+
+    void operator()(const CIEC_BOOL &paQI, const CIEC_STRING &paPARAMS, CIEC_BOOL &paQO, CIEC_STRING &paSTATUS, CIEC_BYTE &paIN) {
```

does it make sense to actually call evt_INIT if this FB is used like a function?

I don't see a use case where this makes sense. I would rather think this should raise an error

---

This is a good point. We should not generate a call-operator when there are multiple events, since it is then not allowed to call the FB without specifying the event in ST anyhow.

---

I have created a corresponding PR for the exporter: https://github.com/eclipse-4diac/4diac-ide/pull/1166